optimisation/ScanSelectionsTree.py

```diff
@@ -308,60 +319,63 @@
+            expBkg = 0.
+            errExpBkg = 0.
             if bkgConfig['isMC']:
-                expBkg = GetExpectedBkgFromMC(hMassBkg, hMassSignal)
-                hMassBkg.Write()
+                expBkg, errExpBkg, hMassBkg = GetExpectedBkgFromMC(hMassBkg, mean, sigma)
+            elif bkgConfig['impFit']:
+                expBkg, errExpBkg, hMassBkg = GetExpectedBkgFromSideBandsImp(hMassBkg, bkgConfig['fitFunc'],
+                                                                             bkgConfig['nSigma'], mean, sigma,
+                                                                             meanSecPeak, sigmaSecPeak)
             else:
-                meanSec = inputCfg['infiles']['secpeak']['mean']
-                sigmaSec = inputCfg['infiles']['secpeak']['sigma']
-                if hMassSecPeak:
-                    expBkg, hMassSB = GetExpectedBkgFromSideBands(hMassBkg, bkgConfig['fitFunc'], bkgConfig['nSigma'],
-                                                                  hMassSignal, -1., -1., hMassSecPeak)
-                elif meanSec > 0 and sigmaSec > 0:
-                    expBkg, hMassSB = GetExpectedBkgFromSideBands(hMassBkg, bkgConfig['fitFunc'], bkgConfig['nSigma'],
-                                                                  hMassSignal, -1., -1., None, meanSec, sigmaSec)
-                else:
-                    expBkg, hMassSB = GetExpectedBkgFromSideBands(hMassBkg, bkgConfig['fitFunc'], bkgConfig['nSigma'],
-                                                                  hMassSignal)
-                hMassSB.Write()
-
+                expBkg, errExpBkg, hMassBkg = GetExpectedBkgFromSideBands(hMassBkg, bkgConfig['fitFunc'],
+                                                                          bkgConfig['nSigma'], mean, sigma,
+                                                                          meanSecPeak, sigmaSecPeak)
+            hMassBkg.Write()
             expBkg *= nExpEv / bkgConfig['nEvents'] / bkgConfig['fractiontokeep']
+            errExpBkg *= nExpEv / bkgConfig['nEvents'] / bkgConfig['fractiontokeep']
 
             if inputCfg['infiles']['background']['corrfactor']['filename']:
                 inFile = TFile.Open(inputCfg['infiles']['background']['corrfactor']['filename'])
                 hBkgCorrFactor = inFile.Get(inputCfg['infiles']['background']['corrfactor']['histoname'])
                 expBkg *= hBkgCorrFactor.GetBinContent(hBkgCorrFactor.FindBin(ptCent))
+                errExpBkg *= hBkgCorrFactor.GetBinContent(hBkgCorrFactor.FindBin(ptCent))
 
             # S/B and significance
             expSoverB = 0.
             expSignif = 0.
-            Serror = 0.
+            errS = 0. # TODO: think how to define a meaningful error on the estimated signal and propagate it
+            errSoverB = 0.
+            errSignif = 0.
             if expBkg > 0:
                 expSoverB = expSignal / expBkg
-            if expSignal + expBkg > 0:
```

Comment:
Hi @fgrosa! Yes, I think you are right. Serror and Berror contain redundant information coming from the expected signal and background. Indeed, I can move this piece of code to the SignificanceScan projector.

Reply:
In the studies for the analysis cut optimization, the interesting information is the errors in the quantities that we estimate. To know how much we can trust what we get. 
I have modified `ProjectSignifNtuple.py` to use the expected uncertainties.

Reply:
@fcatalan92 both are interesting, for different purposes. The uncertainty on the estimate is important to give a feeling about the dependency of the result on our assumptions and modelling, however it is very important also to provide the expected uncertainty given a certain model. In particular, for ITS3 studies this is typically what is quoted, because the point is that given a model, you can expect a given precision and this is what you provide. In addition, the expected uncertainties on signal and background are needed to apply the correct smearing of the input raw yields used in the cut variation method.